crates/bevy_asset/src/io/embedded/embedded_watcher.rs

```diff
@@ -39,7 +39,12 @@ impl EmbeddedWatcher {
             root_paths,
             last_event: None,
         };
-        let watcher = new_asset_event_debouncer(root, debounce_wait_time, handler).unwrap();
+        let watcher = new_asset_event_debouncer(
+            root.to_string_lossy().to_string(),
```

Comment:
I don't think we should silently and lossily convert non-utf8 paths here. In the *vast* majority of cases the paths should be utf-8 and when not, it seems preferable behaviour to panic at the very least.

Reply:
fixed, I replaced this with a `except` so it panics with a message. Same for other places